src/arrayMethodJoin.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = String(this[0]);
+    let separatorVariable = separator;
```

Comment:
You dont need to create additional variable for separator, sipmly use **separator**